backend/src/main/java/de/sambam/gamenighttracker/service/UserService.java

```diff
@@ -102,8 +113,9 @@ public UserService(UserDb userDb) {
         return Optional.empty();
     }
 
-    public Optional<GameSession> addNewGameSession(GameSession newSession, String userId, String apiGameId) {
-        Optional<User> user = userDb.findById(userId);
+    public Optional<GameSession> addNewGameSession(GameSession newSession, String username, String
+            apiGameId) {
+        Optional<User> user = userDb.findByUserName(username);
```

Comment:
Separate the control flow to own methods. This should significantly increase comprehensibility. 🚀